### PR TITLE
Make Dr. CI comment more descriptive

### PR DESCRIPTION
**Pull request by ZainRizvi**
Fixes https://github.com/pytorch/test-infra/issues/4067

Currently the Dr. CI message here loudly proclaims that the PR is failing, when a closer inspection shows that it's merely forked off of a broken trunk.

Change the title of the Dr. CI message so that it's clearer when the failures are unlikely to be the dev's fault

Before:
image

After:
image

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
Nice! This has bugged me for a while.  I hope to see a reduction in the number of force merges due to broken trunk after this

File: torchci/pages/api/drci/drci.ts [hunk 1]
```diff
@@ -194,41 +194,53 @@ export function constructResultsComment(
...
     }
+
+    let title = headerPrefix + icon + ' ' + title_messages.join(', ')
+    output += title
+    output += `\nAs of commit ${sha}:`
+
+    if (!hasAnyFailing) {
+      output += `\n:green_heart: Looks good so far! There are no failures yet. :green_heart:`;
+    }
+    output += constructResultsJobsSections(
```
Comment (huydhn) [inline torchci/pages/api/drci/drci.ts:238]: Is this possible to collapse the BROKEN TRUNK and FLAKY sections to hide them a bit better on Dr. CI?

File: torchci/pages/api/drci/drci.ts [hunk 2]
```diff
@@ -194,41 +194,53 @@ export function constructResultsComment(
...
-
-        if (hasPending) {
-            output += somePending;
-        }
-
-        output += `\nAs of commit ${sha}:`;
-        output += `\n:green_heart: Looks good so far! There are no failures yet. :green_heart:`;
+    const failuresIcon = `:x:`
+    const noneFailing = `No Failures`
+    const significantFailures = `${failedJobs.length} Significant Failures`
```
Comment (malfet) [inline torchci/pages/api/drci/drci.ts:199]: Nit (`1 Significant Failures` feels weird to me)
```suggestion
    const significantFailures = `${failedJobs.length} Significant Failure{failedJobs.length > 1 ? 's' : ''}`
```

**Review by malfet — APPROVED** [review]
Not sure I agree with the world `Significant`, perhaps `New` is better.
Also "English is not German, so One does not need to capitalize every Noun"

File: torchci/pages/api/drci/drci.ts (hunk 1 above)
Comment (ZainRizvi) [reply torchci/pages/api/drci/drci.ts:238]: good idea

**Comment by ZainRizvi** [comment]
> Not sure I agree with the world `Significant`, perhaps `New` is better. Also "English is not German, so One does not need to capitalize every Noun"

I like the idea of using "New"! Sticking with the capitalization though as a matter of taste: Title Case of headings is common enough

**Pull request merged**